### PR TITLE
feat: add burn tax exemption list

### PR DESCRIPTION
**Pull request by nghuyenthevinh2000**
## Summary of changes

Backport to release 1.x, testing is written here: 
* unit test for ante: #54 
* upgrade script test and whitelist update proposal script test: #52 

## Report of required housekeeping

- [ ] Github issue OR spec proposal link
- [x] Wrote tests
- [ ] Updated API documentation (client/lcd/swagger-ui/swagger.yaml)
- [ ] Added a relevant changelog entry: clog add [section] [stanza] [message]

----

## (FOR ADMIN) Before merging

- [ ] Added appropriate labels to PR
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" (coding standards)
- [ ] Confirm added tests are consistent with the intended behavior of changes
- [ ] Ensure all tests pass

### CODE REVIEW

**Review by ZaradarBH — APPROVED** [review]
LGTM :)

File: custom/auth/ante/burntax.go [hunk 1]
```diff
@@ -5,12 +5,14 @@ import (
...
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 	"github.com/cosmos/cosmos-sdk/x/auth/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
 )
 
 // TaxPowerUpgradeHeight is when taxes are allowed to go into effect
 // This will still need a parameter change proposal, but can be activated
 // anytime after this height
 const TaxPowerUpgradeHeight = 9346889
+const WhitelistHeight = 12345678910
```
Comment (inon-man) [inline custom/auth/ante/burntax.go:15]: Don't we have to specify the real height of this?

**Review by inon-man — CHANGES_REQUESTED** [review]

File: custom/auth/ante/burntax.go (hunk 1 above)
Comment (nghuyenthevinh2000) [reply custom/auth/ante/burntax.go:15]: I think it is up to @edk208 to decide

Comment (ZaradarBH) [reply custom/auth/ante/burntax.go:15]: Yes we are still waiting on @edk208 to provide a blockheight for whitelist activation :)

Comment (edk208) [reply custom/auth/ante/burntax.go:15]: do we need a height guard?  wouldn't it just kick in at the time of the upgrade if we use a software upgrade handler?

Comment (nghuyenthevinh2000) [reply custom/auth/ante/burntax.go:15]: yep, it will just kick in at the time of software upgrade

**Review by inon-man — APPROVED** [review]

**Review by inon-man — CHANGES_REQUESTED** [review]
We need to fix burntax test, sorry I am away from the computer now.

**Comment by inon-man** [comment]
@nghuyenthevinh2000 I will update burntax, and burntax_test after #113

**Comment by inon-man** [comment]
@nghuyenthevinh2000 Tests are passed. Please review codes after my commit.

**Comment by nghuyenthevinh2000** [comment]
> We need to fix burntax test, sorry I am away from the computer now.

can you resolve your request change here

**Comment by nghuyenthevinh2000** [comment]
name change for Whitelist: BurnTaxExemptionList

do we concur on this? If so, I will change the name 

@inon-man @ZaradarBH

**Comment by inon-man** [comment]
> name change for Whitelist: BurnTaxExemptionList
> 
> do we concur on this? If so, I will change the name
> 
> @inon-man @ZaradarBH

Yes, and I want it to have more consistency,

ProposalAddWhitelistCmd
ProposalRemoveWhitelistCmd

SetWhitelistAddressProposal
RemoveWhitelistAddressProposal

I think we should stick to Add/Remove semantic.
```
ProposalTypeSetWhitelistAddressProposal    = "SetWhitelistAddressProposal"
ProposalTypeRemoveWhitelistAddressProposal = "RemoveWhitelistAddressProposal"
```
ProposalTypeAddBurnTaxExemptionAddress = "AddBurnTaxExemptionAddress"
ProposalTypeRemoveBurnTaxExemptionAddress = "RemoveBurnTaxExemptionAddress"

like https://github.com/classic-terra/core/blame/v1.x/binance-whitelist/x/treasury/legacy/v04/types.go#L22-L26

**Comment by inon-man** [comment]
@nghuyenthevinh2000 Could you also add a test for remove burn tax exemption list to burn-tax-exemption-test.sh?

**Comment by nghuyenthevinh2000** [comment]
> @nghuyenthevinh2000 Could you also add a test for remove burn tax exemption list to burn-tax-exemption-test.sh?

sure, I expect it should be the same

**Comment by nghuyenthevinh2000** [comment]
@inon-man can you check again

**Review by inon-man — APPROVED** [review]
Great work! @nghuyenthevinh2000 @fragwuerdig

**Pull request merged**